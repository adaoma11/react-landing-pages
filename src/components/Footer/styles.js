import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
