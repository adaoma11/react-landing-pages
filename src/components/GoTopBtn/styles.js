import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
