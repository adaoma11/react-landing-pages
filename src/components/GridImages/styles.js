import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
