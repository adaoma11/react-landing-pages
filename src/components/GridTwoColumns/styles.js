import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 30px;

    @media ${theme.medias.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`;

export const Image = styled.div`
  ${({ $imgSrc }) => css`
    display: block;
    min-height: 500px;
    background-image: url(${$imgSrc});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 60%;
  `}
`;
