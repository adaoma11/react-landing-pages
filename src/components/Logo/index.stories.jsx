import { Logo } from './index';

const meta = {
  title: 'My Components/Logo',
  component: Logo,
  parameters: {
    backgrounds: {
      default: 'light',
    },
  },
};

export default meta;

export const Default = {
  args: {
    text: 'Company Name',
    imgSrc:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQicdrhVQT13rYoFWc8L6SKF8gITrSNtTZpvQ&s',
    href: '#',
  },
};
