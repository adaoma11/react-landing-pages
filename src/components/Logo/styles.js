import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.heading};
    font-size: ${theme.fonts.sizes.large};
    font-weight: bold;
    color: ${theme.colors.primary};
    text-decoration: none;

    > img {
      min-height: 40px;
      max-height: 50px;
      max-width: 150px;
      min-width: 40px;
    }
  `}
`;
