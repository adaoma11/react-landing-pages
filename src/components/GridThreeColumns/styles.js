import styled from 'styled-components';
import { Title as GridItemTitle } from '../Heading/styles';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  counter-reset: grid-counter;
  width: 100%;
  gap: 50px;
`;

export const GridItem = styled.div`
  margin-top: 80px;

  ${GridItemTitle} {
    position: relative;
    left: 4rem;
    margin-bottom: 30px;
  }

  ${GridItemTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 5rem;
    top: -2rem;
    left: -4rem;
    transform: rotate(7deg);
  }
`;
