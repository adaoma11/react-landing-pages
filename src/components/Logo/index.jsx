import PropTypes from 'prop-types';
import * as Styled from './styles';

export const Logo = ({ text, imgSrc, href = '#' }) => {
  if (imgSrc) {
    return (
      <Styled.Container href={href}>
        <img src={imgSrc} alt={text} />
      </Styled.Container>
    );
  }

  return <Styled.Container href={href}>{text}</Styled.Container>;
};

Logo.propTypes = {
  text: PropTypes.string.isRequired,
  imgSrc: PropTypes.string,
  href: PropTypes.string,
};
