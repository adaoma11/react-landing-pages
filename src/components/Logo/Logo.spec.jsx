import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Logo } from '../../components/Logo';

describe('<Logo />', () => {
  it('should have a text', () => {
    renderTheme(<Logo text="texto" href="#" />);
    expect(screen.getByRole('link').firstChild).toHaveTextContent(/texto/i);
  });

  it('should have an image', () => {
    renderTheme(<Logo text="texto" imgSrc="image.png" href="#" />);
    expect(screen.getByRole('img')).toBeInTheDocument();
  });

  it('should have alt equals to text', () => {
    renderTheme(<Logo text="texto" imgSrc="image.png" href="#" />);
    expect(screen.getByRole('img')).toHaveProperty('alt', 'texto');
  });

  it('should have a href', () => {
    renderTheme(<Logo text="texto" href="#" />);
    expect(screen.getByRole('link')).toHaveAttribute('href');
  });

  it('should match snapshot', () => {
    renderTheme(<Logo text="texto" imgSrc="image.png" href="#" />);
    expect(screen.getByRole('link')).toMatchSnapshot();
  });
});
