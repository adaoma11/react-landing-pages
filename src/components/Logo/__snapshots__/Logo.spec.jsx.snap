// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should match snapshot 1`] = `
.c0 {
  font-family: 'Montserrat',sans-serif;
  font-size: 1.5rem;
  font-weight: bold;
  color: #0A1128;
  text-decoration: none;
}

.c0 >img {
  min-height: 40px;
  max-height: 50px;
  max-width: 150px;
  min-width: 40px;
}

<a
  class="c0"
  href="#"
>
  <img
    alt="texto"
    src="image.png"
  />
</a>
`;
